src/modules/accounting/accounting.py

```diff
@@ -174,10 +173,8 @@ def _calculate_report(self, blockstamp: ReferenceBlockStamp):
         # or in the `execute_module` method
         if consensus_version == 1:
             report_data = self._calculate_report_v1(blockstamp)
-        elif consensus_version == 2:
-            report_data = self._calculate_report_v2(blockstamp)
         else:
-            raise IncompatibleException("Consensus version is not supported")
+            report_data = self._calculate_report_latest_version(consensus_version, blockstamp)
```

Comment:
the previous solution was better
- branching happens at the top level
- exception is thrown explicitly if the contract has consensus version > method can handle

consider removing v1 support and leave branching here, smth like:

```python
if consensus_version == 2:
    report_data = self._calculate_report_v2(blockstamp)
elif consensus_version == 3:
    if not is_electra_activated:
        report_data = self._calculate_report_v3_before_electra(blockstamp)
    else:
        report_data = self._calculate_report_v3_after_electra(blockstamp)
else
    raise IncompatibleException("Consensus version is not supported")
```

Reply:
Removed V1 support.
Better to not directly compare to 3. Because there will be a lot of changes, each new consensus version.
Branching at top level makes it hard to implement this thing and keep everything in mind.